From 642902ff7adf9868524dc9e43700548e7102b1f2 Mon Sep 17 00:00:00 2001
From: Luotao Fu <l.fu@pengutronix.de>
Date: Mon, 6 Apr 2009 15:09:16 +0200
Subject: [PATCH] fix core compiling with nmea disabled

struct nmea is only defined in the driver union if NMEA_ENABLED is set.
libpgsd_core however acquires if the ignore_trailing_edge in nmea is set or not
carelessly if NMEA_ENABLED is set. Add a ifdef to prevent this

Signed-off-by: Luotao Fu <l.fu@pengutronix.de>
---
 libgpsd_core.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/libgpsd_core.c b/libgpsd_core.c
index 4c6ca13..654c2f5 100644
--- a/libgpsd_core.c
+++ b/libgpsd_core.c
@@ -201,7 +201,10 @@ static void *gpsd_ppsmonitor(void *arg)
 	    } else if (cycle > 999000 && cycle < 1001000 ) {
 		/* looks like PPS pulse or square wave */
                 if (duration > 499000 && duration < 501000 
-		  && session->driver.nmea.ignore_trailing_edge) {
+#ifdef NMEA_ENABLE
+		  && session->driver.nmea.ignore_trailing_edge
+#endif
+		  ) {
                     /* looks like 1.0 Hz square wave, ignore trailing edge */
                     if (state == 1) {
                          (void)ntpshm_pps(session, &tv);
-- 
1.6.2.1

